bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/AbstractRetryingOperation.java

```diff
@@ -141,7 +144,10 @@ protected void onError(Status status, Metadata trailers) {
     Code code = status.getCode();
     // CANCELLED
     if (code == Status.Code.CANCELLED) {
-      completionFuture.cancel(true);
+      if (!completionFuture.isDone()) {
+        // cancel(true) would trigger GrpcFuture.interuptTask().  We don't want that to happen.
+        completionFuture.cancel(false);
```

Comment:
I'm going to revert to cancel(true).  I was worried about this invocation of cancel() cascading to another RPC cancel.  `GrpcFuture.cancel(true)` calls `GrpcFuture.interruptTask()` which calls `AbstractRetryingOperation.cancel()` which does an RPC cancel.  However, it turns out that in this case `AbstractRetryingOperation.cancel()` should be a no-op.  I'll ensure that it's a no-op by extending the synchronized block in `onOK()`